docs/src/lib/NavLink.svelte

```diff
@@ -1,12 +1,17 @@
-<script>
+<script lang="ts">
+  import type { Snippet } from "svelte";
+  import type { HTMLAnchorAttributes } from "svelte/elements";
   import { page } from "$app/stores";
 
-  export let href = "";
+  interface Props {
+    children: Snippet;
+    href: string;
+  }
 
-  /** @type {"page"|undefined} */
-  $: ariaCurrent = $page.url.pathname === href ? "page" : undefined;
+  let { href = "", children, ...rest }: Props & HTMLAnchorAttributes = $props();
+  let ariaCurrent = $derived($page.url.pathname === href ? ("page" as const) : undefined);
 </script>
 
-<a {href} aria-current={ariaCurrent} {...$$restProps}>
-  <slot />
+<a {href} aria-current={ariaCurrent} {...rest}>
+  {@render children?.()}
```

Comment:
~~What's the reason for dropping `<slot />` in favour of `{@render children?.()}`?~~

_Update_: I've read the docs now. I'm quite sure I need some time to get used to drop `<slot />` but I see it's deprecated. So: nice. Soon I'll be able to see the benefits of the new snippet feature in Svelte 5.